ee/desktop/runner/runner.go

```diff
@@ -350,6 +350,51 @@ func (r *DesktopUsersProcessesRunner) killDesktopProcesses(ctx context.Context)
 	)
 }
 
+// killDesktopProcess kills the existing desktop process for the given uid
+func (r *DesktopUsersProcessesRunner) killDesktopProcess(ctx context.Context, uid string) error {
+	proc, ok := r.uidProcs[uid]
+	if !ok {
+		return fmt.Errorf("could not find desktop proc for uid %s, cannot kill process", uid)
+	}
+
+	// unregistering client from runner server so server will not respond to its requests
+	r.runnerServer.DeRegisterClient(uid)
+
+	client := client.New(r.userServerAuthToken, proc.socketPath)
+	err := client.Shutdown(ctx)
+	if err == nil {
+		r.slogger.Log(ctx, slog.LevelInfo,
+			"killed user desktop process",
+			"uid", uid,
+		)
+		delete(r.uidProcs, uid)
+		return nil
+	}
+
+	// We didn't successfully send a shutdown request -- check to see if it's because
+	// the process is already gone.
+	if !r.processExists(proc) {
+		delete(r.uidProcs, uid)
+		return nil
+	}
+
+	r.slogger.Log(ctx, slog.LevelWarn,
+		"failed to send shutdown command to user desktop process, killing process instead",
+		"uid", uid,
+		"pid", proc.Process.Pid,
+		"path", proc.path,
+		"err", err,
+	)
+
+	if err := proc.Process.Kill(); err != nil {
+		return fmt.Errorf("could not kill desktop process for uid %s with pid %d: %w", uid, proc.Process.Pid, err)
+	}
+
+	// Successfully killed process
+	delete(r.uidProcs, uid)
+	return nil
```

Comment:
worth adding an affirmative log here to match the `"killed user desktop process"` logging behavior above?

Reply:
Yes! Updated